tests/e2e/utils/auto-rds-s3-cleanup.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+from time import sleep
+import boto3
+
+from utils import get_rds_client, get_s3_client, get_secrets_manager_client
+
+
+def main():
+    metadata_dict = {}
+    with open("auto-rds-s3-setup-metadata", "r") as setup_metadata:
+        for line in setup_metadata.readlines():
+            metadata = line.split("=")
+            metadata_dict[metadata[0]] = metadata[1].strip("\n")
+    secrets_manager_client = get_secrets_manager_client(CLUSTER_REGION)
+    delete_s3_bucket(metadata_dict, secrets_manager_client)
+    delete_rds(metadata_dict, secrets_manager_client)
+
+
+def delete_s3_bucket(metadata_dict, secrets_manager_client):
+    s3_client = get_s3_client(CLUSTER_REGION)
+    s3_resource = boto3.resource("s3")
+    bucket_name = metadata_dict["bucket_name"]
+
+    print("Deleting S3 bucket...")
+
+    bucket = s3_resource.Bucket(bucket_name)
+    bucket.objects.all().delete()
+    s3_client.delete_bucket(Bucket=bucket_name)
+
+    buckets = s3_client.list_buckets()["Buckets"]
+    assert any(bucket["Name"] == bucket_name for bucket in buckets) is False
+    print("S3 bucket has been successfully deleted")
+
+    secrets_manager_client.delete_secret(
+        SecretId=metadata_dict["s3_secret_name"], ForceDeleteWithoutRecovery=True
+    )
+
+    try:
+        secrets_manager_client.describe_secret(SecretId=metadata_dict["s3_secret_name"])
+
+    except:
+        print("S3 secret has been successfully deleted")
```

Comment:
-->
```
secrets_manager_client.delete_secret(
        SecretId=metadata_dict["s3_secret_name"], ForceDeleteWithoutRecovery=True
    )
print("S3 secret has been successfully deleted")
```

the describe secret call will likely still describe the deleted secret with the secret status being 'DELETED' or something similar, so the try catch doesn't help here

as long as the request to delete goes through we can assume the secret is deleted